src/planwise/client/projects/handlers.cljs

```diff
@@ -194,26 +194,81 @@
                     (assoc-in [:facilities :isochrones] nil))
          filters (db/project-filters new-db)
          project-id (get-in db [:project-data :id])]
-     (api/update-project project-id filters :facilities
-                         :projects/project-updated)
-     new-db)))
+     (transition-map-state new-db :facilities))))
+
+(defn transition-map-state [db request-with]
+  (let [new-db
+        (case (get-in db [:map-state :current])
+          :loaded            (loaded-to-request-pending db)
+          :request-pending   (request-pending-to-request-pending db)
+          :loading           (loading-to-request-pending db)
+          :loading-displayed (loading-displayed-to-request-pending db))]
+    (-> new-db
+        (assoc-in [:map-state :current] :request-pending)
+        (assoc-in [:map-state :request-with] request-with))))
+
+(defn loaded-to-request-pending [db]
+  (set-request-timeout db))
+
+(defn request-pending-to-request-pending [db]
+  (-> db (cancel-prev-timeout) (set-request-timeout)))
+
+(defn loading-to-request-pending [db]
+  (-> db (cancel-prev-timeout) (cancel-prev-request) (set-request-timeout)))
+
+(defn loading-displayed-to-request-pending [db]
+  (-> db (cancel-prev-request) (set-request-timeout)))
+
+(defn set-request-timeout [db]
+  (assoc-in db [:map-state :timeout]
+            (js/setTimeout #(dispatch [:projects/trigger-map-request]) 500)))
+
+(defn cancel-prev-timeout [db]
+  (js/clearTimeout (get-in db [:map-state :timeout]))
+  (assoc-in db [:map-state :timeout] nil))
```

Comment:
`(update-in db [:map-state :timeout] js/clearTimeout)` is more readable and does exactly the same here.